VERSION-RANGE-SPEC.rst

```diff
@@ -0,0 +1,989 @@
+apply to the comparators of any two contiguous constraints to be valid:
+
+- "!=" constraint can be followed by a constraint using any comparator, i.e.,
+  any of "=", "!=", ">", ">=", "<", "<=" as comparator (or no constraint).
+
+Ignoring all constraints with "!=" comparators:
+
+- A "=" constraint must be followed only by a constraint with one of "=", ">",
+  ">=" as comparator (or no constraint).
+
+And ignoring all constraints with "=" or "!=" comparators, the sequence of
+constraint comparators must be an alternation of greater and lesser comparators:
+
+- "<" and "<=" must be followed by one of ">", ">=" (or no constraint).
+- ">" and ">=" must be followed by one of "<", "<=" (or no constraint).
+
+Tools must report an error for such invalid ranges.
+
+
+Parsing and validating version range specifiers
+-------------------------------------------------
+
+To parse a version range specifier string:
+
+- Remove all spaces and tabs.
+- Start from left, and split once on colon ":".
+- The left hand side is the URI-scheme that must be lowercase.
+  - Tools must validate that the URI-scheme value is ``vers``.
+- The right hand side is the specifier.
+
+- Split the specifier from left once on a slash "/".
+
+- The left hand side is the <versioning-scheme> that must be lowercase.
+  Tools should validate that the <versioning-scheme> is a known scheme.
+
+- The right hand side is a list of one or more constraints.
+  Tools must validate that this constraints string is not empty ignoring spaces.
+
+- If the constraints string is equal to "*", the <version-constraint>is "*".
+  Parsing is done and no further processing is needed for this ``vers``. A tool
+  should report an error if there are extra characters beyond "*". 
+
+- Strip leading and trailing pipes "|" from the constraints string.
+- Split the constraints on pipe "|". The result is a list of  <version-constraint>.
+  Consecutive pipes must be treated as one and leading and trailing pipes ignored.
+
+- For each <version-constraint>:
+  - Determine if the <version-constraint> starts with one of the two comparators:
+
+    - If it starts with ">=", then the comparator is ">=".
+    - If it starts with "<=", then the comparator is "<=".
+    - If it starts with "!=", then the comparator is "!=".
+    - If it starts with "<",  then the comparator is "<".
+    - If it starts with ">",  then the comparator is ">".
+
+    - Remove the comparator from <version-constraint> string start. The
+      remaining string is the version.
+
+  - Otherwise the version is the full <version-constraint> string (which implies
+    an equality comparator of "=")
+
+  - Tools should validate and report an error if the version is empty.
+
+  - If the version contains a percent "%" character, apply URL quoting rules
+    to unquote this string.
+
+  - Append the parsed (comparator, version) to the constraints list.
+
+Finally:
+
+- The results are the <versioning-scheme> and the list of <version, comparator>
```

Comment:
As the operands are not commutable it feels weird to see them in reverse order...
```suggestion
- The results are the ``<versioning-scheme>`` and the list of ``<comparator, version>``